src/pl/michaeli/service/ConfirmationResource.java

```diff
@@ -0,0 +1,220 @@
+		int screeningId = reservation.getScreeningId();
+		List<BookedSeat> seatsToBook = reservation.getBookedSeats();
+
+		if(name.length()<3 || surname.length()<3) {
+			ErrorMessage errorMessage = new ErrorMessage("Za kr�tkie imi� lub nazwisko!");
+            return Response.serverError().type(MediaType.APPLICATION_JSON).entity(errorMessage).build();
+		}
+		if(checkName(name)==false) {
+			ErrorMessage errorMessage = new ErrorMessage("B��dne imi�!");
+            return Response.serverError().type(MediaType.APPLICATION_JSON).entity(errorMessage).build();
+		}
+		if(checkSurname(surname)==false) {
+			ErrorMessage errorMessage = new ErrorMessage("B��dne nazwisko!");
+            return Response.serverError().type(MediaType.APPLICATION_JSON).entity(errorMessage).build();
+		}
+		if(seatsToBook==null || seatsToBook.size()==0) {
+			ErrorMessage errorMessage = new ErrorMessage("Trzeba wybra� minimum jedno miejsce!");
+			return Response.serverError().type(MediaType.APPLICATION_JSON).entity(errorMessage).build();
+		}
+		if(!checkIsAvailable(seatsToBook,screeningId)) {
+			ErrorMessage errorMessage = new ErrorMessage("Wybrano zaj�te miejsce!");
+			return Response.serverError().type(MediaType.APPLICATION_JSON).entity(errorMessage).build();
+		}
+		if(checkIfFreeSpaceLeft(seatsToBook,screeningId)) {
+			ErrorMessage errorMessage = new ErrorMessage("Nie mo�na zostawi� jednego miejsca wolnego mi�dzy dwoma zaj�tymi!");
+			return Response.serverError().type(MediaType.APPLICATION_JSON).entity(errorMessage).build();
+		}
+		
+		seatsToBook.forEach((seatToBook) -> seatToBook.setScreeningId(screeningId));
+
+		AnnotationConfigApplicationContext context = new AnnotationConfigApplicationContext(AppConfig.class);
+		ScreeningsDAO screeningsDAO=context.getBean(ScreeningsDAO.class);
+		
+		Date screeningDate = (screeningsDAO.getScreeningById(screeningId)).getScreeningDate();
+		reservation.setExpirationTime(this.countExpirationDate(screeningDate));
+		reservation.setTotalAmount(this.countTotalAmount(seatsToBook,screeningDate));
+
+		ReservationsDAO reservationsDAO=context.getBean(ReservationsDAO.class);
+		BookedSeatsDAO bookedSeatsDAO=context.getBean(BookedSeatsDAO.class);
+
+		reservation=reservationsDAO.addReservation(reservation);
+		bookedSeatsDAO.addBookedSeats(reservation);
+		reservation.setBookedSeats(bookedSeatsDAO.getBookedSeatsByReservationId(reservation.getReservationId()));
+
+		context.close();
+
+        GenericEntity<Reservation> myEntity = new GenericEntity<Reservation>(reservation) {};
+        return Response.status(200).entity(myEntity).build();
+    }
+		
+	private Boolean checkName(String name) {
+		Boolean isCorrect = Character.isUpperCase(name.charAt(0));
+        for(int i=1;i<name.length();i++) {
+            if(isCorrect) {
+                isCorrect = Character.isLowerCase(name.charAt(i));
+            }
+        }
+		return isCorrect;
+	}
+	
+	private Boolean checkSurname(String surname) {
+		if(surname.contains("-")) {
```

Comment:
A co jeśli zawiera dwa myślniki?

Reply:
Rzuci błędem :)

Reply:
Sprawdzasz to gdzieś indziej? Nie widzę nigdzie, ale może kiepsko patrzę..

Reply:
wewnątrz if'a dzielę input na 2 części (przed '-' i po '-') i potem każda jest jeszcze sprawdzana funkcją checkName(String name) - chyba nazwa trochę myląca w tym wypadku..